src/xenia/kernel/xam/content_manager.cc

```diff
@@ -259,6 +264,29 @@ std::filesystem::path ContentManager::ResolveGameUserContentPath() {
   return root_path_ / title_id / kGameUserContentDirName / user_name;
 }
 
+const std::vector<object_ref<XFile>> ContentManager::GetOpenedFilesFromContent(
+    const std::string_view root_name) {
+  std::vector<object_ref<XFile>> opened_files = {};
+  const std::vector<object_ref<XFile>> all_files_handles =
+      kernel_state_->object_table()->GetObjectsByType<XFile>(
+          XObject::Type::File);
+
+  std::string resolved_path = "";
+  kernel_state_->file_system()->FindSymbolicLink(std::string(root_name) + ':',
+                                                 resolved_path);
+
+  for (const object_ref<XFile>& file : all_files_handles) {
+    std::filesystem::path file_path =
+        std::filesystem::path(file->entry()->absolute_path());
+
+    auto symlinked_path = file_path.parent_path().string() + '\\';
```

Comment:
Would this `\` separator work on Linux, and is it possible that `resolved_path` won't have one? I'd avoid making any assumptions about the string form of a path, and path == string comparisons, since one path can be represented by many different strings, with the trailing `\` or `/` present or not present (while still being a valid directory path), with `.` and `..`.

Reply:
Paths here are paths from VFS, so they're independent from platform. (like ``\\Device\\content\\1``)

Reply:
That sounds like an even worse case for std::filesystem::path usage, if I understand correctly, it can do normalization for the host OS internally; also comparison is case-sensitive here in this place, is that good?

Reply:
std::filesystem::path overall should be avoided for guest paths, as they have Xbox 360's path format, not the host OS format. The VFS code of Xenia uses ResolvePath (accepting a raw string path) in many places, it's preferable to perform all operations with guest paths in terms of VFS objects, not std::filesystem::path or strings.

Reply:
I was worried about both of these cases too.
``std::filesystem::path`` seems to work correctly for these cases. I haven't seen anything that would do some trickery to tinker with host os, but would be nice to be 100% sure. 
Case sensitive isn't an issue here, because both paths are based on the same data (initial symlink path) and they must match exactly.

Reply:
`file_path.parent_path()` may become `/Device/content/1` instead of `\Device\content\1`.

Reply:
ohhh right... Ok I will change it

Reply:
Changed, but I made quite bold assumption about existing of ``\`` character in string. I guess we shouldn't check it as w/o one symlink wouldn't even exist nor file 🤔